listen/array-changes.js

```diff
@@ -104,7 +110,7 @@ var observableArrayProperties = {
             array_sort.apply(this, arguments);
 
             // dispatch after change events
-            for (var i = 0; i < this.length; i++) {
+            for (i = 0, countI = this.length; i < countI; i++) {
```

Comment:
I’ve used the pattern elsewhere: `for (var index = 0, length = this.length; index < length; index++)` in keeping with using full and meaningful names.